library/asn1parse.c

```diff
@@ -149,10 +149,16 @@ int mbedtls_asn1_get_int( unsigned char **p,
     if( ( ret = mbedtls_asn1_get_tag( p, end, &len, MBEDTLS_ASN1_INTEGER ) ) != 0 )
         return( ret );
 
-    if( len == 0 || len > sizeof( int ) || ( **p & 0x80 ) != 0 )
+    if( len == 0 || len > sizeof( int ) )
         return( MBEDTLS_ERR_ASN1_INVALID_LENGTH );
 
-    *val = 0;
+    if( ( ret = mbedtls_check_shortest_asn1_integer( *p, len ) ) != 0 )
+        return( ret );
+
+    if( **p & 0x80 )
```

Comment:
Blocker: As far as I see, we're subsequently treating the sign bit as a value bit here.